src/puppetlabs/puppetdb/cli/util.clj

```diff
@@ -0,0 +1,41 @@
+(ns puppetlabs.puppetdb.cli.util
+  "As this namespace is required by both the tk and non-tk subcommands,
+  it must remain very lightweight, so that subcommands like
+  \"version\" aren't slowed down by loading the entire logging
+  subsystem or trapperkeeper, etc."
+  (:require
+   [puppetlabs.i18n.core :refer [trs]]))
+
+(def err-exit-status 1)
+
+;; Testing hook
+(defn java-version [] (System/getProperty "java.version"))
+
+(defn jdk-support-status [version]
+  "Returns :official, :tested, or :unknown, or :no."
+  (cond
+    (re-matches #"1.[123456]($|(\..*))" version) :no
+    (re-matches #"1.7($|(\..*))" version) :unknown
+    (re-matches #"1.8($|(\..*))*" version) :official
+    (re-matches #"1.10($|(\..*))*" version) :tested
+    :else :unknown))
+
+(defn jdk-unsupported-msg [version]
+  (let [status (jdk-support-status version)]
+    (case status
+      (:official :tested :unknown) nil
```

Comment:
Would it be difficult to log a warning when the java version is untested since we don't want to load the logging libraries?

Reply:
Hmm, first let me consider what I think about warning here, i.e. is that common, and/or what we'd want regardless, etc.

Reply:
(I think we decided for now to make sure the docs are clear, and then maybe revisit this question later, if we like...)